tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -5642,15 +5644,20 @@ Status ConvertFullyConnectedHelper(OpConverterParams* params,
       params->converter, TRT_TensorOrWeights(output_tensor), output_dim,
       /*validation_only=*/false, &output_tensor, node_def,
       /*op_instance=*/1));
-  params->outputs->push_back(TRT_TensorOrWeights(output_tensor));
+  if (output) {
+    *output = output_tensor;
+  } else {
+    params->outputs->push_back(TRT_TensorOrWeights(output_tensor));
+  }
   *converted = true;
   return Status::OK();
 }
 
 Status ConvertMatMulHelper(OpConverterParams* params,
                            TRT_TensorOrWeights input_a,
                            TRT_TensorOrWeights input_b, bool transpose_a,
-                           bool transpose_b) {
+                           bool transpose_b,
+                           nvinfer1::ITensor** output = nullptr) {
```

Comment:
Let's make two routines:
StatusOr<nvinfer1::ITensor*> ConvertMatMulImpl(...) doesn't add the output to params->outputs.
Status ConvertMatMulHelper(...)  calls convertMatMulImpl and add the output to params->outputs.